benefits/core/models/enrollment.py

```diff
@@ -306,14 +276,12 @@ def clean(self):
                 if not template_path(t):
                     template_errors.append(ValidationError(f"Template not found: {t}"))
 
-        if field_errors:
-            raise ValidationError(field_errors)
         if template_errors:
             raise ValidationError(template_errors)
 
     def eligibility_form_instance(self, *args, **kwargs):
         """Return an instance of this flow's EligibilityForm, or None."""
-        if not bool(self.eligibility_form_class):
+        if not self.uses_api_verification:
```

Comment:
This is nice, it's more explicit and also checks on the value of `self.eligibility_api_url`